src/busctl/busctl.c

```diff
@@ -1373,7 +1373,12 @@ static int monitor(int argc, char **argv, int (*dump)(sd_bus_message *m, FILE *f
                 if (r > 0)
                         continue;
 
-                r = sd_bus_wait(bus, UINT64_MAX);
+                r = sd_bus_wait(bus, arg_timeout > 0 ? arg_timeout : UINT64_MAX);
```

Comment:
> What I wanted to achieve was being able to say "wait for one matching message, but at most for some time", and the current implementation does this.

and to be clear, what I wanted to have was, being able in a script to wait for exactly the first match and then exit, or timeout if no match appears, so that I don't need to muck about with loops or manual timers or whatnot. And again, no problem if there's another option that has different semantics if people want that, no objections whatsoever.